hikari/errors.py

```diff
@@ -227,27 +214,30 @@ class HTTPResponseError(HTTPError):
     raw_body: typing.Any = attr.ib()
     """The response body."""
 
-    reason: typing.Optional[str] = attr.ib(default=None)
-    """The error reason. If `builtins.None`, will generate one automatically."""
-
-    message: str = attr.ib(default=_message_default_factory, init=False)
+    message: str = attr.ib(default="")
     """The error message."""
 
-    def __str__(self) -> str:
-        try:
-            raw_body = self.raw_body.decode("utf-8")
-        except (AttributeError, UnicodeDecodeError):
-            raw_body = str(self.raw_body)
-
-        chomped = len(raw_body) > 200
+    code: int = attr.ib(default=0)
+    """The error code."""
 
+    def __str__(self) -> str:
         if isinstance(self.status, http.HTTPStatus):
             name = self.status.name.replace("_", " ").title()
             name_value = f"{name} {self.status.value}"
         else:
             name_value = str(self.status).title()
 
-        return f"{name_value}: {raw_body[:200]}{'...' if chomped else ''} for {self.url}"
+        if self.message:
+            body = self.message
+        else:
+            try:
+                body = self.raw_body.decode("utf-8")
+            except (AttributeError, UnicodeDecodeError):
```

Comment:
This is probably a non-issue but seeing as raw_body is typed as any, we don't have any guarantee that if it has a "decode" method it'll be compatible with that call signature, was wondering if it might be worth it also catching TypeError and ValueError here to cover every case

Reply:
Will add them for good measure :+1: